modal/io_streams.py

```diff
@@ -35,22 +37,28 @@ async def _sandbox_logs_iterator(
 
 
 async def _container_process_logs_iterator(
-    process_id: str, file_descriptor: int, last_entry_id: Optional[str], client: _Client
-):
+    process_id: str, file_descriptor: int, client: _Client
+) -> AsyncIterator[Optional[str]]:
     req = api_pb2.ContainerExecGetOutputRequest(
         exec_id=process_id,
         timeout=55,
-        last_batch_index=last_entry_id or 0,
         file_descriptor=file_descriptor,
     )
     async for batch in client.stub.ContainerExecGetOutput.unary_stream(req):
         if batch.HasField("exit_code"):
-            yield (None, batch.batch_index)
+            yield None
             break
         for item in batch.items:
-            # TODO: do this on the server.
-            if item.file_descriptor == file_descriptor:
-                yield (item.message, batch.batch_index)
+            yield item.message
+
+
+class StreamType(Enum):
+    # Discard all logs from the stream.
+    DEVNULL = 0
```

Comment:
To be foot-gun-safe, you could reuse the values of the subprocess equivalents here (they are also int valued afaik)
e.g. `DEVNULL = subprocess.DEVNULL`

Then, even though we expose our own constant, things wouldn't break if a user unknowingly used subprocess.DEVNULL instead etc. Minor but could save us some support issues in the future :)